registry/job.go

```diff
@@ -316,9 +365,81 @@ func (r *EtcdRegistry) DestroyUnit(name string) error {
 	}
 
 	// TODO(jonboulle): add unit reference counting and actually destroying Units
+
```

Comment:
The problem is that the delete operation needs to be atomic (essentially a
compare and delete, "delete this unit file IFF no other units reference
it"), but that information is not stored in one place in etcd at the
moment. That's where a reference count would come in.

I haven't looked at the PR code yet but - how does it deal with this
scenario of a race between destroying a unit and creating a new unit with
an identical unit file hash?

On Fri, Mar 18, 2016 at 2:00 PM Djalal Harouni <notifications@github.com>
wrote:

> In registry/job.go
> <https://github.com/coreos/fleet/pull/1510#discussion_r56651729>:
>
> > @@ -316,9 +365,81 @@ func (r *EtcdRegistry) DestroyUnit(name string) error {
> >  	}
> >
> >  	// TODO(jonboulle): add unit reference counting and actually destroying Units
> > +
>
> @jonboulle <https://github.com/jonboulle> could you please remember what
> was the use case that made you add this comment ? "unit reference counting"
> is there some optimization involved or something else ? thanks :-)
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/coreos/fleet/pull/1510/files/0be1141333baa76d27b0560f96614a371350a663#r56651729>
>